packages/eslint-plugin/src/rules/no-unnecessary-condition.ts

```diff
@@ -184,6 +185,7 @@ export default createRule<Options, MessageId>({
         'Unnecessary conditional, the types have no overlap.',
       noStrictNullCheck:
         'This rule requires the `strictNullChecks` compiler option to be turned on to function correctly.',
+      suggestRemoveOptionalChain: 'Replace `?.` with `.`',
```

Comment:
[Nit]: I think the message is not always accurate as the suggestion fixer can sometimes remove `?.` and not replace it with `.` ([deploy preview playground link](https://deploy-preview-11081--typescript-eslint.netlify.app/play/#ts=5.8.2&fileType=.tsx&code=CYUwxgNghgTiAEAzArgOzAFwJYHtVJxwAoBKALngDcctgBuAKAYHpmksAPEYeLVDHPAAGiQqSENROAPwA6UoxZtEnbr36ChAbQCMAGngAmAwGYAuloAMZiboPH45uVbOMgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0yHJgBNK%2BSpPRRE0aB2iRwYAL4gtQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)):

```ts
declare function foo(): void;

// fixed into `foo()`
foo?.();

// fixed into `[1, 2, 3][0]`
[1, 2, 3]?.[0];
```

I couldn't think of a better alternative though (and I think the existing message explains the suggestion fixer pretty well).

Reply:
Yeah, this crossed my mind while working on it... I agree with you your suggestion!

```suggestion
      suggestRemoveOptionalChain: 'Remove unnecessary optional chain',
```